cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/resource/AzureVolumeResourceBuilder.java

```diff
@@ -155,9 +157,12 @@ private String getAvailabilityZone(AuthenticatedContext auth, CloudResource vm)
             DeviceNameGenerator generator = new DeviceNameGenerator();
             futures.addAll(volumeSet.getVolumes().stream()
                     .map(volume -> intermediateBuilderExecutor.submit(() -> {
-                        Disk result = client.createManagedDisk(
-                                volume.getId(), volume.getSize(), AzureDiskType.getByValue(
-                                        volume.getType()), region, resourceGroupName, cloudStack.getTags());
+                        Disk result = client.getDiskByName(resourceGroupName, volume.getId());
```

Comment:
Can you add a logline that the disk already exists and we're not doing anything.